oximeter/db/src/lib.rs

```diff
@@ -393,4 +395,90 @@ mod tests {
         assert!(TimeseriesName::try_from("a:").is_err());
         assert!(TimeseriesName::try_from("123").is_err());
     }
+
+    #[test]
+    fn test_timeseries_key_generation_hashes_fields_sequentially() {
+        use super::timeseries_key_for;
+        use oximeter::{Field, FieldValue};
+
+        let f = |name: &str, value| Field { name: name.to_string(), value };
+
+        // Confirm that "targets" and "metrics" are interchangeable,
+        // we just hash everything sequentially.
+        assert_eq!(
+            timeseries_key_for(
+                [&f("a", FieldValue::String("a".to_string()))].into_iter(),
+                [&f("b", FieldValue::String("b".to_string()))].into_iter(),
+            ),
+            timeseries_key_for(
+                [
+                    &f("a", FieldValue::String("a".to_string())),
+                    &f("b", FieldValue::String("b".to_string())),
+                ]
+                .into_iter(),
+                [].into_iter(),
+            ),
+        );
+
+        // However, order still matters ("a, b" != "b, a")
+        assert_ne!(
+            timeseries_key_for(
+                [&f("a", FieldValue::String("a".to_string()))].into_iter(),
+                [&f("b", FieldValue::String("b".to_string()))].into_iter(),
+            ),
+            timeseries_key_for(
+                [&f("b", FieldValue::String("b".to_string()))].into_iter(),
+                [&f("a", FieldValue::String("a".to_string()))].into_iter(),
+            ),
+        );
+    }
```

Comment:
Kind of a weird test. I don't know that we buy too much by essentially confirming that `timeseries_key_for` concats the argument vecs in the order they're given.

Reply:
Agreed, I think I wanted this as a proxy for "is the `timeseries_key` for a `Sample` stable?"

We already have the tests for individual fields, but that doesn't tell us anything about "what if you have a sample that tries to change the order of fields?"

I went ahead and updated this test, to simply `EXPECTORATE` a `timeseries_key` value for a simple sample. That's the actual property I want to validate, so IMO it makes for a better test: 929e761c9210b501fe83126936eca85bbddc83de